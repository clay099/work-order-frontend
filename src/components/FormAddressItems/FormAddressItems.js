import React from "react";
import { TextField, Grid } from "@material-ui/core";

const FormAddressItems = ({ formData, handleChange }) => {
	return (
		<>
			<Grid item xs={12} sm={6}>
				<TextField
					variant="outlined"
					required
					fullWidth
					id="streetAddress"
					label="Street Address"
					name="streetAddress"
					onChange={handleChange}
					value={formData.streetAddress}
				/>
			</Grid>
			<Grid item xs={12} sm={6}>
				<TextField
					variant="outlined"
					required
					fullWidth
					id="zip"
					name="zip"
					label="Zip Code"
					onChange={handleChange}
					value={formData.zip}
				/>
			</Grid>
			<Grid item xs={12} sm={6}>
				<TextField
					variant="outlined"
					required
					fullWidth
					id="city"
					name="city"
					label="City"
					onChange={handleChange}
					value={formData.city}
				/>
			</Grid>
			<Grid item xs={12} sm={6}>
				<TextField
					variant="outlined"
					required
					fullWidth
					id="country"
					name="country"
					label="Country"
					onChange={handleChange}
					value={formData.country}
				/>
			</Grid>
		</>
	);
};

export default FormAddressItems;
