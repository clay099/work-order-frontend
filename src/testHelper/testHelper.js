// testHelper.js
import React from "react";
import { render as rtlRender } from "@testing-library/react";
import { createStore } from "redux";
import { Provider } from "react-redux";
// Import your own reducer
import reducer from "../reducers/rootReducer";

function render(
	ui,
	{
		initialState,
		store = createStore(reducer, initialState),
		...renderOptions
	} = {}
) {
	function Wrapper({ children }) {
		return <Provider store={store}>{children}</Provider>;
	}
	return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

const tokenState = { token: "token" };
const userTypeUserState = { userType: "user" };
const userTypeTradesmenState = { userType: "user" };
const idState = { id: 1234 };
const userDetailsState = { userDetails: "userDetails" };
const handleSubmitState = { handleSubmit: jest.fn() };
const handleChangeState = { handleChange: jest.fn() };
const disabledState = { disabled: false };
const disabledEmailState = { disabledEmail: false };
const buttonTextState = { buttonText: "buttonText" };
const formDataState = {
	formData: {
		id: 1,
		firstName: "firstName",
		lastName: "lastName",
		email: "email",
		password: "password",
		newPassword: "newPassword",
		confirmNewPassword: "confirmNewPassword",
	},
};

// re-export everything
export * from "@testing-library/react";
// override render method
export {
	render,
	tokenState,
	userTypeUserState,
	userTypeTradesmenState,
	idState,
	userDetailsState,
	handleSubmitState,
	handleChangeState,
	disabledState,
	disabledEmailState,
	buttonTextState,
	formDataState,
};
