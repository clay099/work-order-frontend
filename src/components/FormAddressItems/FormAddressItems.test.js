import React from "react";
import FormAddressItems from "./FormAddressItems";
import { render } from "../../testHelper/testHelper";
console.error = jest.fn();

const params = {
	formData: {
		city: "city",
		streetAddress: "streetAddress",
		zip: "zip",
		country: "country",
	},
	handleChange: jest.fn,
	disabled: true,
};

describe("FormAddressItems Component", () => {
	// smoke test
	test("renders without crashing", () => {
		render(<FormAddressItems {...params} />);
	});

	// snapshot test
	test("matches snapshot", () => {
		const { asFragment } = render(<FormAddressItems {...params} />);
		expect(asFragment).toMatchSnapshot();
	});

	test("value to be in the document", () => {
		const { getByText } = render(<FormAddressItems {...params} />);
		const value = getByText("Street Address");
		expect(value).toBeInTheDocument();
	});
});
