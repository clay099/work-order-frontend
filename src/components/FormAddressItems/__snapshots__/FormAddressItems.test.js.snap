// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormAddressItems Component matches snapshot 1`] = `[Function]`;
